source/multigrid/mg_transfer_prebuilt.cc

```diff
@@ -284,23 +284,43 @@ void MGTransferPrebuilt<VECTOR>::build_matrices (
   // We keep track in the bitfield dof_touched which global dof has
   // been processed already (on the current level). This is the same as
   // the multigrid running in serial.
-  // Only entering on the finest level gives wrong results (why?)
+
+  struct dof_pair
```

Comment:
on a separate note -- I don't think you can use a locally declared class as a template argument in C++98, because it does not have external linkage. I think you need to pull it out of the function. But I'm only 80% sure on that.